### PR TITLE
chore(ci): run policy tests against various Gateway API versions

### PR DESCRIPTION
**Pull request by olix0r**
This change expands the policy test matrix so that:

* The Minimum Supported Kubernetes Version is tested with the Minimum Supported
  Gateway API Version.
* The latest Kubernetes version is tested with the latest Gateway API version.
* The latest Kubernetes version is tested with the CLI's vendored Gateway API
  types. This is near the MSGV, though currently provides experimental types.

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -162,13 +162,25 @@ jobs:
...
+      # minimum Gateway API version.
       matrix:
         k8s:
-          - v1.23
           - v1.32
-        linkerd-managed-gw-crds:
-          - true
-          - false
+        gateway-api:
+          - version: v1.2.1
```
Comment (olix0r) [inline .github/workflows/integration.yml:172]: Actually, we probably want to test the MSGV here as well (with only the standard subset).

```suggestion
          - version: v1.1.1
```

because we already cover the latest experimental featureset.

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -163,9 +163,24 @@ jobs:
...
+          - version: linkerd
+            channel: experimental
+          - version: v1.3.0-rc.1
+            channel: experimental
+        # Also test the Minimum Supported Kubernetes Version with the Minimum
+        # Supported Gateway API version.
+        include:
+          - k8s: v1.23
+            version: v1.1.1
+            channel: standard
```
Comment (olix0r) [inline .github/workflows/integration.yml:180]:
```suggestion
          - k8s: v1.23
            gateway-api:
              version: v1.1.1
              channel: standard
```
